### PR TITLE
Add libspdm_set_data for provisioned_slot_id

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #1518

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

### CODE REVIEW

File: doc/user_guide.md [hunk 1]
```diff
@@ -386,7 +391,12 @@ Refer to spdm_server_init() in [spdm_responder.c](https://github.com/DMTF/spdm-e
    }
    ```
 
-   1.6, if PSK is required, optionally deploy PSK Hint.
+   1.6, if public key of responder was provisoned for requester, the provisioned_slot_id is which slot_id cert_chain was provsioned for requester.
+   ```
+   libspdm_set_data (spdm_context, LIBSPDM_DATA_LOCAL_PUBLIC_CERT_CHAIN_DEFAULT_SLOT_ID, NULL, provisioned_slot_id, sizeof(provisioned_slot_id));
+   ```
+
+   1.7, if PSK is required, optionally deploy PSK Hint.
```
Comment (jyao1) [inline doc/user_guide.md:399]: This text is confusing. I suggest no update, so far.

We have another big patch to support raw public key provision. Will discuss later.

Please drop the user_guide update in this patch.

Comment (Wenxing-hou) [reply doc/user_guide.md]: Thanks. I have deleted the user_guide change.

**Comment by steven-bellock** [comment]
So before this change folks had to provision the raw public key (leaf certificate) in slot 0? Now they can choose any slot?

**Comment by jyao1** [comment]
> So before this change folks had to provision the raw public key (leaf certificate) in slot 0? Now they can choose any slot?

Right. It is a bug.

**Comment by steven-bellock** [comment]
Why's that a bug? Why would they want to provision the raw public key in, for example, slot 4?

**Comment by jyao1** [comment]
how can you assume that public key is also from slot 0?

**Comment by steven-bellock** [comment]
On the Responder's side, these slots aren't visible to the Requester, since `CERT_CAP==0`. When raw public keys are used libspdm is just leveraging slots to make things easier. So the Integrator shouldn't care which internal "slot" libspdm stores the public key in.

**Comment by jyao1** [comment]
The integrator of requester does not care, which real slot is used by when slot 0xFF is used, because the public cert is already provisioned. That is correct.

But the integrator of the responder needs to know that, because it must choose a right public cert, when slot 0xFF is used, to **create the transcript**. It cannot choose arbitrary public cert, because it will fail the transcript hash.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**